tmc-langs-util/src/main/java/fi/helsinki/cs/tmc/langs/util/TaskExecutor.java

```diff
@@ -0,0 +1,22 @@
+package fi.helsinki.cs.tmc.langs.util;
+
+import com.google.common.base.Optional;
+import fi.helsinki.cs.tmc.langs.ExerciseDesc;
+import fi.helsinki.cs.tmc.langs.RunResult;
+import fi.helsinki.cs.tmc.stylerunner.validation.ValidationResult;
+import java.nio.file.Path;
+
+
+public interface TaskExecutor {
+
+    public void prepareSolution(Path path);
+
+    public void prepareStub(Path path);
+
+    public Optional<ValidationResult> runCheckCodeStyle(Path path);
+
+    public Optional<RunResult> runTests(Path path);
+
+    public Optional<ExerciseDesc> scanExercise(Path path, String exerciseName);
```

Comment:
I'd rather have these throw a specific exception if the language plugin could not be found instead of returning an absent and having a vague error message.